sdk/eventgrid/azure-eventgrid/README.md

```diff
@@ -1,21 +1,122 @@
-# Microsoft Azure SDK for Python
+# Azure EventGrid client library for Python
 
-This is the Microsoft Azure Event Grid Client Library.
-This package has been tested with Python 2.7, 3.5, 3.6, 3.7 and 3.8.
-For a more complete view of Azure libraries, see the [azure sdk python release](https://aka.ms/azsdk/python/all).
+Azure Event Grid is a fully-managed intelligent event routing service that allows for uniform event consumption using a publish-subscribe model.
 
+[Source code][python-eg-src] | [Package (PyPI)][python-eg-pypi] | [API reference documentation][python-eg-ref-docs]| [Product documentation][python-eg-product-docs] | [Samples][python-eg-samples]
 
-# Usage
+## Getting started
 
-For code examples, see [Event Grid](https://docs.microsoft.com/python/api/overview/azure/event-grid)
-on docs.microsoft.com.
+### Prerequisites
+* Python 2.7, or 3.5 or later is required to use this package.
+* You must have an [Azure subscription][azure_subscription] and an EventGrid Topic resource to use this package.
 
+### Install the package
+Install the Azure EventGrid client library for Python with [pip][pip]:
 
-# Provide Feedback
+```bash
+pip install azure-eventgrid
+```
 
-If you encounter any bugs or have suggestions, please file an issue in the
-[Issues](https://github.com/Azure/azure-sdk-for-python/issues)
-section of the project.
+#### Create an EventGrid Topic
+You can create the resource using [Azure Portal][azure_portal_create_EG_resource]
 
+### Authenticate the client
+In order to interact with the Eventgrid service, you will need to create an instance of a client. 
+A **topic_hostname** and **credential** are necessary to instantiate the client object.
 
-![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-python%2Fazure-eventgrid%2FREADME.png)
+#### Looking up the endpoint
+You can find the endpoint and the hostname on the Azure portal.
+
+#### Create the client with AzureKeyCredential
+
+To use an API key as the `credential` parameter,
+pass the key as a string into an instance of [AzureKeyCredential][azure-key-credential].
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid import EventGridPublisherClient
+
+topic_hostname = "https://<name>.<region>.eventgrid.azure.net/api/events"
+credential = AzureKeyCredential("<api_key>")
+eg_publisher_client = EventGridPublisherClient(topic_hostname, credential)
+```
+
+## Key concepts
+
+### EventGridPublisherClient
+`EventGridPublisherClient` provides operations to send event data to topic hostname specified during client initialization.
+Either a list or a single instance of CloudEvent/EventGridEvent/CustomEvent can be sent.
+
+### EventGridConsumer
+`EventGridConsumer` is used to desrialize an event received.
+
+## Troubleshooting
```

Comment:
In leiu of anything else in this section, add a single bullet for enabling the proper logger to get our internal traces.

```suggestion
## Troubleshooting

- Enable `azure.eventgrid` logger to collect traces from the library
```

Reply:
fixed